servers/src/common/adapters.rs

```diff
@@ -238,9 +245,11 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 			return false;
 		}
 
-		for header in bhs.iter() {
-			for hook in &self.hooks {
-				hook.on_header_received(&header, &addr);
+		if !self.sync_state.is_syncing() {
```

Comment:
This `headers_received` is only ever called during sync - so we can just skip this one entirely for the purpose of the hooks. Normal "header first" block propagation would go through the single `header_received` above.

Reply:
Thanks, removed this block altogeather